src/main/java/com/example/bookshop/dto/CreateBookRequestDto.java

```diff
@@ -0,0 +1,20 @@
+package com.example.bookshop.dto;
+
+import com.example.bookshop.validation.Isbn;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+public record CreateBookRequestDto(@NotNull
+                                   String title,
+                                   @NotNull
+                                   String author,
+                                   @NotNull
+                                   @Min(0)
+                                   BigDecimal price,
+                                   @Isbn
+                                   @NotNull
+                                   String isbn,

```

Comment:
If you'd like to make it more simple you can validate isbn by adding Pattern annotation instead of your custom Isbn annotation. In Pattern ann you can directly spicafy validation regex and message in case of invalid isbn. But it's your choice whether to leave it like this or not, if you want more custom control on your validation